### ISSUE TITLE
SQL_CALC_FOUND_ROWS not supported

### ISSUE THREAD

**Issue by ghost** [vitessio/vitess#3904]
There is a syntax error when parsing the SQL like followed, it looks like 'SQL_CALC_FOUND_ROWS' not supported 

```SQL
SELECT  SQL_CALC_FOUND_ROWS  *  FROM  `tb`  WHERE  `col1`  =  "a"  and  `col2`  = 105 LIMIT 1
```

**Comment by morgo** [comment]
It is easy to add this syntax to the parser, but the problem is that this query is usually followed up with a query like `SELECT FOUND_ROWS()`.

Because Vitess uses a connection pool, it is not guaranteed that the followup query will use the same connection unless a transaction is currently open.

Implementation will require Vitess automatically issuing a followup `SELECT FOUND_ROWS()` if the parser detected a `SQL_CALC_FOUND_ROWS` and then storing it in thread-local storage. Queries that then issue this select can be returned directly from vtgate. This is a similar problem to https://github.com/vitessio/vitess/issues/3668

**Comment by morgo** [comment]
This query is required for Wordpress to install on Vitess:
```
SELECT SQL_CALC_FOUND_ROWS  wp_posts.ID FROM wp_posts  
WHERE 1=1  AND wp_posts.post_type = 'post' AND (wp_posts.post_status = 'publish')  ORDER BY wp_posts.post_date DESC LIMIT 0, 10;
```

**Comment by rajveerappan** [comment]
FYI, FOUND_ROWS is deprecated as of MySQL 8.0.17:
https://dev.mysql.com/doc/refman/8.0/en/information-functions.html#function_found-rows
> The SQL_CALC_FOUND_ROWS query modifier and accompanying FOUND_ROWS() function are deprecated as of MySQL 8.0.17 and will be removed in a future MySQL version. As a replacement, considering executing your query with LIMIT, and then a second query with COUNT(*) and without LIMIT to determine whether there are additional rows.

**Comment by systay** [comment]
FYI - being worked in #5732

**Comment by systay** [comment]
Fixed by #6680
